src/realm/object-store/c_api/object.cpp

```diff
@@ -0,0 +1,531 @@
+#include <realm/object-store/c_api/types.hpp>
+#include <realm/object-store/c_api/util.hpp>
+
```

Comment:
Compiling without sync fails:
```C++
cmake -DCMAKE_BUILD_TYPE=debug -DREALM_ENABLE_SYNC=0 -DREALM_NO_TESTS=1 ..
cmake --build . -j8
```
fails with 
```
make[1]: *** [src/realm/object-store/c_api/CMakeFiles/RealmFFIStatic.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/Users/nabil/Dev/realm/KMP/master-1/realm-kotlin/external/core/src/realm/object-store/c_api/object.cpp:361:29: error: no member named 'overloaded' in
      namespace 'realm::util'
        auto getter = util::overloaded{
                      ~~~~~~^
/Users/nabil/Dev/realm/KMP/master-1/realm-kotlin/external/core/src/realm/object-store/c_api/object.cpp:361:39: error: expected ';' at end of declaration
        auto getter = util::overloaded{
```
unless we add explicitly the following import:

```C++
#include <realm/util/overloaded.hpp>
```
cc @rorbech

Reply:
I've added the include, should pass CI momentarily.